packages/api/src/controllers/animalController.js

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Model, transaction } from 'objection';
+import AnimalModel from '../models/animalModel.js';
+import baseController from './baseController.js';
+
+const animalController = {
+  getFarmAnimals() {
+    return async (req, res) => {
+      try {
+        const { farm_id } = req.headers;
+        const rows = await AnimalModel.query().where({ farm_id });
+        return res.status(200).send(rows);
+      } catch (error) {
+        console.error(error);
+        return res.status(500).json({
+          error,
+        });
+      }
+    };
+  },
+
+  addAnimal() {
+    return async (req, res) => {
+      try {
+        const trx = await transaction.start(Model.knex());
+        const { farm_id } = req.headers;
+        const result = [];
+
+        if (!Array.isArray(req.body)) {
+          return res.status(400).send('Request body should be an array');
+        }
+
+        for (const animal of req.body) {
+          if (!animal.identifier && !animal.name) {
+            return res.status(400).send('Should send either animal name or identifier');
+          }
+
+          if (!animal.default_breed_id && !animal.custom_breed_id) {
```

Comment:
Should we also check that the custom_breed_id belongs to that farm?

Reply:
Yes, good call!

Reply:
We might also want the check the other constraint if both id types exist on one animal :( -- otherwise you get a model validation error which is much less user friendly. 

I wish we had TS on our backend because these checks can get out of hand on how many are needed haha